### PR TITLE
feat: add signup process

### PR DESCRIPTION
**Pull request by LBokan**
### Description

Added createUser modal, connected it to the server

### Type of change

- [x] Feature (introduces new functionality)
- [ ] Bugfix
- [ ] Configuration
- [ ] Refactoring (doesn't change functionality, only the code)
- [ ] Style changes (doesn't affect functionality, only the look of the components)

### PR Checklist

- [x] Synced up with latest from the main branch/Code is linted
- [x] Description of the changes has been added (Screenshots/Video)
- [x] Material UI components/constants/hooks have been used wherever possible

> If not explain here

- [x] Commit messages follow [conventional commits guideline](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#summary)
- [x] Console has been checked for warnings

#### Screenshots/Video before change

Add screenshots of the changed components here.

#### Screenshots/Video after change

SignUp

SignUp-modal

https://user-images.githubusercontent.com/117748123/207851329-aac2860e-be33-412a-953d-c8823b324e6f.mp4

### CODE REVIEW

File: src/client/api/mutation/user.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { gql } from '@apollo/client';
+
+export const CREATE_USER = gql`
```
Comment (ahodkov) [inline src/client/api/mutation/user.js]: Думаю, что более правильно: SIGNUP/REGISTRATION

**Review by ahodkov — CHANGES_REQUESTED** [review]

File: src/client/api/mutation/user.js (hunk 1 above)
Comment (LBokan) [reply src/client/api/mutation/user.js]: Изменила на REGISTRATION

File: src/client/api/mutation/user.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { gql } from '@apollo/client';
 
-export const CREATE_USER = gql`
+export const REGISTRATION = gql`
   mutation createUser($email: String!, $password: String!) {
```
Comment (ahodkov) [inline src/client/api/mutation/user.js]: createUser надо тоже заменить везде

**Review by ahodkov — CHANGES_REQUESTED** [review]

File: src/client/api/mutation/user.js (hunk 2 above)
Comment (LBokan) [reply src/client/api/mutation/user.js]: Заменила

File: src/client/components/RegistrationModal/RegistrationModal.jsx [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+              right: '10px'
+            }}
+            onClick={formik.dirty ? openConfirmationModal : resetFormOnClose}
+            aria-label="Close create device modal"
+          >
+            <Close />
+          </IconButton>
+
+          <Typography variant="h2" sx={{ mb: '35px', fontSize: '18px' }}>
+            Add new user
```
Comment (ahodkov) [inline src/client/components/RegistrationModal/RegistrationModal.jsx]: Registration/Signup

File: src/client/components/RegistrationModal/RegistrationModal.jsx [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+            alt="Waves image"
+          />
+        </Stack>
+      </Modal>
+
+      <ConfirmationModal
+        isOpen={isOpenConfirmationModal}
+        onClose={closeConfirmationModal}
+        onSubmit={resetFormOnClose}
+        text="Do you really want to cancel the adding of a new user? All you entered data will be lost"
```
Comment (ahodkov) [inline src/client/components/RegistrationModal/RegistrationModal.jsx]: the adding of a new user -> the registration

File: src/client/pages/Login.jsx [hunk 5]
```diff
@@ -27,41 +45,120 @@ export const Login = () => {
...
+        <NotificationBar
+          text={errorRegistration.message}
+          typeOfBar="error"
+          setIsClose={setIsOpenRegistrationError}
+        />
+      )}
+
+      {!!isOpenRegistrationSuccess && (
+        <NotificationBar
+          text={'User was created successfully'}
```
Comment (ahodkov) [inline src/client/pages/Login.jsx]: Registration was successfully

File: src/client/components/RegistrationModal/RegistrationModal.jsx (hunk 3 above)
Comment (LBokan) [reply src/client/components/RegistrationModal/RegistrationModal.jsx]: Исправила

File: src/client/components/RegistrationModal/RegistrationModal.jsx (hunk 4 above)
Comment (LBokan) [reply src/client/components/RegistrationModal/RegistrationModal.jsx]: Исправила

File: src/client/pages/Login.jsx (hunk 5 above)
Comment (LBokan) [reply src/client/pages/Login.jsx]: Исправила

**Review by ahodkov — APPROVED** [review]

**Pull request merged**